### DISCUSSION TITLE
Where notices get rendered

### DISCUSSION THREAD

**Discussion by szepeviktor** [docsifyjs/docsify#2389] [Q&A]
On the docsify-themeable demo page I see notices

![image](https://github.com/docsifyjs/docsify/assets/952007/ceb8b8e5-a9de-41df-b4ca-6eed7b7136fd)

```markdown
!> **Important** notice with `inline code` and additional placeholder text used
to force the content to wrap and span multiple lines.

?> **Tip** notice with `inline code` and additional placeholder text used to
force the content to wrap and span multiple lines.
```

but cannot find the code that renders this Markdown to HTML.
Could you help me?

**Comment by TongyiDai** [comment]
In docsify 4.13.1 this is handled in the paragraph renderer, not a separate notice plugin.

The relevant source is:

https://github.com/docsifyjs/docsify/blob/v4.13.1/src/core/render/compiler/paragraph.js

It checks the rendered paragraph text for the escaped prefixes:

`/^!&gt;/` → `helperTpl('tip', text)`

`/^\?&gt;/` → `helperTpl('warn', text)`

Then `src/core/render/tpl.js` implements the helper as a paragraph with the corresponding class:

`<p class="tip">…</p>` or `<p class="warn">…</p>`

The theme CSS supplies the visual treatment. The `content.slice(5)` in the helper removes the HTML-escaped marker (`!&gt;` or `?&gt;`) before inserting the remaining content.

One version caveat: the current v5 branch has moved callout handling to `src/core/render/compiler/blockquote.js` and recognizes blockquote-style markers such as `[!TIP]`. If you are tracing the demo that used the original `!>` / `?>` syntax, inspect the v4.13.1 tag rather than `develop`, otherwise the implementation will look unrelated.
